compiler/src/dotty/tools/dotc/transform/Splicer.scala

```diff
@@ -242,6 +243,11 @@ object Splicer {
       case Literal(Constant(value)) =>
         interpretLiteral(value)
 
+      case tree: Ident if tree.symbol.is(Inline, butNot = Method) =>
+        tree.tpe.widenTermRefExpr match
+          case ConstantType(c) => c.value.asInstanceOf[Object]
```

Comment:
We should probably handle `null` here too

Reply:
We do not support `inline val n = null`. That should be added when we add support for it and can write a test case for that logic.

Reply:
See #12198